### DISCUSSION TITLE
Build failed for Ruby 3.2.0-dev: Incorrect bison version

### DISCUSSION THREAD

**Discussion by michaelmelanson** [rbenv/ruby-build#2129] [Build failures] [answered]
I just tried to install ruby 3.2.0-dev, and it failed with this error:

```
BUILD FAILED (macOS 12.6.1 using ruby-build 20221025)

Inspect or clean up the working tree at /var/folders/43/dmhknl5j7dvf6jn5ynzdkm5c0000gn/T/ruby-build.20230118144513.75854.wiIsQA
Results logged to /var/folders/43/dmhknl5j7dvf6jn5ynzdkm5c0000gn/T/ruby-build.20230118144513.75854.log

Last 10 log lines:
making ractor.rbinc
compiling random.c
compiling range.c
compiling rational.c
compiling re.c
compiling regcomp.c
compiling regenc.c
parse.tmp.y:12.10-14: require bison 3.0, but have 2.3
make: *** [parse.c] Error 63
make: *** Waiting for unfinished jobs....
```

I tried installing bison via homebrew with `brew install bison`, which installed bison version 3.8.3. This didn't change the output, so I assume it's still using a system version.

**Comment by hsbt** [comment]
You can see its detail via `brew info bison`.

>bison is keg-only, which means it was not symlinked into /opt/homebrew

**Reply by mislav** [reply]
@hsbt Our wiki currently says:
> Ruby 3.2.0-dev and above (only `-dev` versions) require Bison 3+: `brew install bison`

1. Is it true that only `-dev` versions of Ruby require Bison, or is it every Ruby 3.2+ that requires it?
2. If Homebrew bison is keg-only, then only doing `brew install bison` will not be enough for Ruby compilation. Should ruby-build auto-detect Homebrew bison and configure Ruby for it?

**Reply by hsbt** [reply]
> 1. Is it true that only `-dev` versions of Ruby require Bison

That is. bison is only required `-dev` version that build from raw git repository.

**Reply by hsbt** [reply]
FYI: I'm working an official requirements document at https://github.com/ruby/ruby/blob/master/doc/contributing/building_ruby.md

**Comment by minimul** [comment]
I'm on Big Sur and this worked:

```
brew install bison # installs bison version 3.8.4
PATH="/usr/local/opt/bison/bin:$PATH" asdf install ruby 3.2.1
```

- asdf uses ruby-build
- brew installs bison as keg only hence the need to temporarily modify the PATH variable.

**Reply by whilekofman** [reply]
Tried this but getting this error.
```zsh: command not found: asdf```
I use rbenv but even when I replace asdf with rbenv 

> I'm on Big Sur and this worked:
> 
> ```
> brew install bison # installs bison version 3.8.4
> PATH="/usr/local/opt/bison/bin:$PATH" asdf install ruby 3.2.1
> ```
> 
> * asdf uses ruby-build
> * brew installs bison as keg only hence the need to temporarily modify the PATH variable.

**Reply by mislav** [reply]
I don't think you need Bison at all for Ruby 3.2.1 (at least not in my experience)
```sh
brew install libyaml readline gmp openssl@1.1  # also openssl@3 works too
RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1) --disable-install-doc" rbenv install 3.2.1
```

**Reply by Rim-777** [reply]
> I'm on Big Sur and this worked:
> 
> ```
> brew install bison # installs bison version 3.8.4
> PATH="/usr/local/opt/bison/bin:$PATH" asdf install ruby 3.2.1
> ```
> 
> * asdf uses ruby-build
> * brew installs bison as keg only hence the need to temporarily modify the PATH variable.

Works for me for RVM

**Comment by mislav** [comment, marked as answer]
Apart from installing Bison, you also need to **explicitly configure your PATH** so that Homebrew-provided bison takes precedence over system bison:

```sh
brew install bison libyaml gmp readline openssl@1.1
```
```sh
PATH="$(brew --prefix bison)/bin:$PATH" \
  RUBY_CONFIGURE_OPTS="--with-openssl-dir=$(brew --prefix openssl@1.1) --disable-install-doc" \
  rbenv install 3.2.0-dev
```

Note: only `-dev` versions of Ruby require Bison. The PATH adjustment is _not_ necessary for stable versions.

**Reply by whilekofman** [reply]
Hi I did try to install 3.2.1 and it continued giving me the error.  I tried doing the path install multiple times and it kept erroring out.  Eventually I opened a new terminal window and ran it again which is when it got past the  bison problem and errored on the libyaml issue which has been more common but was simple to resolve. 

Thanks for all your help!
